aws/resource_aws_iam_saml_provider.go

```diff
@@ -113,8 +113,14 @@ func resourceAwsIamSamlProviderDelete(d *schema.ResourceData, meta interface{})
 		SAMLProviderArn: aws.String(d.Id()),
 	}
 	_, err := conn.DeleteSAMLProvider(input)
+	if err != nil {
+		if isAWSErr(err, iam.ErrCodeNoSuchEntityException, "") {
```

Comment:
Nit: If we're adding new code, we can use `tfawserr.ErrCodeEquals()`/`tfawserr.ErrMessageContains()` to save the later refactoring. 👍